### ISSUE TITLE
Deploy the platform atlas app

### ISSUE THREAD

**Issue by laineymajor** [department-of-veterans-affairs/va.gov-team#62854]
### PROBLEM STATEMENT
Right now the app lives on Eric's server lol... we need to move it to a more appropriate location :) 

### ACTION ITEMS
- [x] determine which env to deploy this to
- [x] create new argo app (Lindsey onboarded to utility)

##### Argo App Onboarding:
- [x] Onboard app to utility

##### Chart/Manifest checklist:
- [x] External DNS
- [x] HPA for server
- [x] Ingress Routes
- [x] Server Deployment 
- [x] Server Service
- [x] Worker (Sidekiq) Deployment
- [x] HPA for worker
- [x] Service Account
- [x] Secrets, env vars, etc (waiting on a few things for this)

#### Devops tasks
- [x] External DNS for platform-codeowners.vfs.va.gov
- [x] [Service Account](https://github.com/department-of-veterans-affairs/devops/pull/13347)
- [x] [ECR Repo](https://github.com/department-of-veterans-affairs/devops/pull/13348)
- [x] Create svc account/iam access key [See doc](https://vfs.atlassian.net/wiki/spaces/OT/pages/2516746289/Add+new+service+account+with+ECR+permissions)
- [x] Postgres (Kyle is working on this)
- [x] Redis (Kyle is working on this)
- [x] Specific param store path to store our values e.g. /platform-codeowners/utility/
- [x] Add AWS_SECRET_ACCESS_KEY and AWS_ACCESS_KEY_ID to Github GHA secrets

**Comment by LindseySaari** [comment]
Follow these steps for onboarding the new app to EKS: https://vfs.atlassian.net/wiki/spaces/OT/pages/1474594142/Onboard+a+new+application+to+EKS

**Comment by LindseySaari** [comment]
Argo App onboarding has been started here: https://github.com/department-of-veterans-affairs/va.gov-team/issues/62306

Which [created this PR](https://github.com/department-of-veterans-affairs/vsp-infra-argocd/pull/211/files), but I think we want it to be the utility env, not dev

**Comment by LindseySaari** [comment]
platform-codeowners has been deployed to the utility cluster: https://argocd.vfs.va.gov/applications/platform-codeowners-utility?view=tree&resource= 

Working with Kyle to offboard from dev.


Next steps:
- Write manifests
   - Deploy rails app
   - Deploy sidekiq
- Create dependencies:
   - Postgres (RDS)
   - Redis (Elasticache)
   - ECR repo
   - Service Account

**Comment by LindseySaari** [comment]
Started on a branch here for the manifest files:

https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/tree/codeowners-deployment

**Comment by laineymajor** [comment]
Should these items live in this ticket?

1. Datadog APM for the app
2. DNS name for the app (do we want this behind socks)
3. Github Actions related to the deployment (ECR build, push and image tag update) - I think this could be a separate ticket
4. Do we want to integrate Sidekiq UI into the app to debug failing jobs
5. Extract secrets from the app and implement the secrets.yaml/param store flow

**Comment by LindseySaari** [comment]
Made some really great progress on this. I have all the manifest files in place, but we need the devops steps to take place before deploying the app. I will continue to pick at items that I can work on. I need to comb through the values and templates and update the secrets with values as they're obtained.

**Comment by LindseySaari** [comment]
Made some really great progress on this. I have all the manifest files in place, but we need the devops steps to take place before deploying the app. I will continue to pick at items that I can work on. I need to comb through the values and templates and update the secrets with values as they're obtained.

**Comment by laineymajor** [comment]
Kyle completed hand off with Chris on Friday... Chris to start taking on this work
- next step: elasticache!

**Comment by laineymajor** [comment]
Got elastiche up; been debugging.

**Comment by laineymajor** [comment]
@flooose working on remaining 404... working with hector and rachal
- look at the main terraform utility -> database has not been setup for the IM

**Comment by laineymajor** [comment]
- The login issue was fixed by @RachalCassity .... this was due to the ssl
- reimporting the roster on deployment to eric's infra
- seed job is failing because of a mismatch
- need to get PGDump working @flooose to assist

**Comment by LindseySaari** [comment]
This is completed! We have the app deployed and the sidekiq healthchecks are now working!
